app/scripts/modules/core/src/pipeline/config/triggers/triggers.directive.js

```diff
@@ -51,8 +53,25 @@ module.exports = angular.module('spinnaker.core.pipeline.config.trigger.triggers
       }
 
       pipeline.triggers
-        .forEach(t => t.expectedArtifactIds = t.expectedArtifactIds
-          .filter(eid => expectedArtifact.id !== eid));
+        .forEach(t => {
+          if (t.expectedArtifactIds) {
+            t.expectedArtifactIds = t.expectedArtifactIds.filter(eid => expectedArtifact.id !== eid);
+          }
+        });
+
+      if (!pipeline.stages || pipeline.stages.length === 0) {
+        return;
+      }
+
+      pipeline.stages.forEach(stage => {
+        artifactReferenceService.walkReferences('stage', stage, (obj, key) => {
+          if (Array.isArray(obj)) {
+            obj.splice(key, 1);
+          } else {
+            delete obj[key];
```

Comment:
This looks like it will delete the selected artifact regardless of whether it's the artifact we deleted

Reply:
oy.  that'll teach me for only testing with a single expected artifact.  thankyou.